flink-cdc-composer/src/main/java/com/ververica/cdc/composer/utils/FactoryDiscoveryUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.composer.utils;
+
+import com.ververica.cdc.common.factories.Factory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/** Discovery utilities for {@link Factory}. */
+public class FactoryDiscoveryUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(FactoryDiscoveryUtils.class);
+
+    private FactoryDiscoveryUtils() {}
+
+    /** Returns the {@link Factory} for the given identifier. */
+    @SuppressWarnings("unchecked")
+    public static <T extends Factory> T getFactoryByIdentifier(
+            String identifier, Class<T> implementClass) {
```

Comment:
What about using `factoryClass` instead of `implementClass` for the second argument? The function uses a factory interface for searching an implementation.